test/controllers/api/v2/hosts_controller_test.rb

```diff
@@ -724,9 +724,8 @@ def test_rebuild_tftp_config
       Setting[:restrict_registered_smart_proxies] = false
       SETTINGS[:require_ssl] = false
 
-      Resolv.any_instance.stubs(:getnames).returns(['else.where'])
+      @controller.expects(:auth_smart_proxy).never
       post :facts, params: { :name => hostname, :facts => facts }
-      assert_nil @controller.detected_proxy
```

Comment:
Isn't this still a good assertion?

Reply:
:shrug: well the two assertions together lead me to believe I should look into the `auth_smart_proxy` method why is it assigned, when it should not. So I believe it is missleading assertion and assertion for the method not to be called is enough and is much clearer to what code path the test is actually testing.